src/promptflow/promptflow/contracts/flow.py

```diff
@@ -583,7 +583,10 @@ def is_normal_node(self, node_name):
 
     def is_llm_node(self, node):
         """Given a node, return whether it uses LLM tool."""
-        return node.type == ToolType.LLM
+        if isinstance(node, Node):
+            return node.type == ToolType.LLM
+        else:
+            raise TypeError(f"Unexpected node type {type(node)}")
```

Comment:
Why change this? Is there a possible reason why the node is not of type `Node`?

Reply:
The GPT generated test code remind me of this corner case. If it is not necessary, I'll remove the change in the next commit.

Reply:
I don't think it is necessary but not sure, you can double confirm with @thy09.